cilium-cli/clustermesh/clustermesh.go

```diff
@@ -1594,52 +1596,97 @@ func (k *K8sClusterMesh) validateCAMatch(aiLocal, aiRemote *accessInformation) (
 // (certgen) mode. As with classic mode, only autodetected IP-based
 // clustermesh-apiserver Service endpoints are currently supported.
 func (k *K8sClusterMesh) ConnectWithHelm(ctx context.Context) error {
-	localRelease, err := getRelease(k.client.(*k8s.Client), k.params)
+	localClient := k.client.(*k8s.Client)
+	localRelease, err := getRelease(localClient, k.params)
 	if err != nil {
 		k.Log("❌ Unable to find Helm release for the target cluster")
 		return err
 	}
-
-	localClient, remoteClient, err := k.getClientsForConnect()
+	remoteClients, err := k.getRemoteClients()
 	if err != nil {
 		return err
 	}
 
-	aiLocal, aiRemote, err := k.getAccessInfoForConnect(ctx, localClient, remoteClient)
+	localHelmValues := localRelease.Config
+	localOldClusters, err := getOldClusters(localHelmValues)
 	if err != nil {
 		return err
 	}
-
-	err = k.validateInfoForConnect(aiLocal, aiRemote)
+	aiLocal, err := k.getAccessInfoForConnect(ctx, localClient, k.params.SourceEndpoints)
 	if err != nil {
 		return err
 	}
 
-	// Validate that CA certificates match between the two clusters
-	match, err := k.validateCAMatch(aiLocal, aiRemote)
-	if err != nil {
-		return err
-	} else if !match {
-		k.Log("⚠️ Cilium CA certificates do not match between clusters. Multicluster features will be limited!")
-	}
+	localNewClusters := make([]map[string]interface{}, 0)
+	remoteHelmValuesAll := make([]map[string]interface{}, 0)
+	remoteHelmValuesBD := make([]map[string]interface{}, 0)
+	remoteOldClustersAll := make([][]map[string]interface{}, 0)
+	remoteNewClusters := make([]map[string]interface{}, 0)
+	remoteClusterNames := make([]string, 0)
+	airemoteClusterNames := make([]string, 0)
+	for _, remoteClient := range remoteClients {
+		remoteClusterNames = append(remoteClusterNames, remoteClient.ClusterName())
+		aiRemote, err := k.getAccessInfoForConnect(ctx, remoteClient, k.params.DestinationEndpoints)
+		airemoteClusterNames = append(airemoteClusterNames, aiRemote.ClusterName)
+		if err != nil {
+			return err
+		}
 
-	// Get existing helm values for the local cluster
-	localHelmValues := localRelease.Config
-	// Expand those values to include the clustermesh configuration
-	localHelmValues, err = updateClustermeshConfig(localHelmValues, aiRemote, !match)
-	if err != nil {
-		return err
+		err = k.validateInfoForConnect(aiLocal, aiRemote)
+		if err != nil {
+			return err
+		}
+
+		// Validate that CA certificates match between the two clusters
+		match, err := k.validateCAMatch(aiLocal, aiRemote)
+		if err != nil {
+			return err
+		} else if !match {
+			k.Log("⚠️ Cilium CA certificates do not match between clusters. Multicluster features will be limited!")
```

Comment:
Do we need to continue here or should this be an error?

Reply:
> Do we need to continue here or should this be an error?

We need to continue. It's just a warning. It's already on the main branch:
https://github.com/cilium/cilium/blob/main/cilium-cli/clustermesh/clustermesh.go#L1623